core/challenges.go

```diff
@@ -40,7 +40,9 @@ func HTTPChallenge01(accountKey *jose.JsonWebKey) Challenge {
 
 // TLSSNIChallenge01 constructs a random tls-sni-00 challenge
 func TLSSNIChallenge01(accountKey *jose.JsonWebKey) Challenge {
-	return newChallenge(ChallengeTypeTLSSNI01, accountKey)
+	chall := newChallenge(ChallengeTypeTLSSNI01, accountKey)
+	chall.N = 25
```

Comment:
As for the number of ports to validate, we probably also want that to be a parameter here, with a field in the Challenge object that is not exposed to the web.  Cf. `prepareChallengeForDisplay()` in `web-front-end.go`.